File: Gamblers Ascent/src/Card.cpp
#include "Card.h"

auto Card::to_string(Card::suit_t s) -> std::string
{
	switch (s)
	{
	case suit_t::hearts:
		return "hearts";
	case suit_t::diamonds:
		return "diamonds";
	case suit_t::clubs:
		return "clubs";
	case suit_t::spades:
		return "spades";
	}
}

auto Card::get_value() -> unsigned {
	switch (rank)
	{
	case rank_t::ace:
		return 11;
	case rank_t::two:
		return 2;
	case rank_t::three:
		return 3;
	case rank_t::four:
		return 4;
	case rank_t::five:
		return 5;
	case rank_t::six:
		return 6;
	case rank_t::seven:
		return 7;
	case rank_t::eight:
		return 8;
	case rank_t::nine:
		return 9;
	case rank_t::ten:
		return 10;
	case rank_t::jack:
		return 10;
	case rank_t::queen:
		return 10;
	case rank_t::king:
		return 10;
	}
}


int operator+(const Card& c1, const Card& c2)
{
	return static_cast<int>(c1.rank) + static_cast<int>(c2.rank);
}

std::ostream& operator<<(std::ostream& os, Card& c)
{
	os << "The " << c.get_value() << " of " << c.to_string(c.suit) << '\n';
	return os;
}

File: Assets/Scripts/Blackjack/Card.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public enum Suit
{
    Spades = 1,
    Clubs = 2,
    Diamonds = 3,
    Hearts = 4
}

public enum Rank
{
    Ace = 1,
    Two = 2,
    Three = 3,
    Four = 4,
    Five = 5,
    Six = 6,
    Seven = 7,
    Eight = 8,
    Nine = 9,
    Ten = 10,
    Jack = 11,
    Queen = 12,
    King = 13
}

public class Card 
{
    public Suit Suit { get; }
    public Rank Rank { get; }
    
    // Card's sprite image gotten from the suit and rank using the image's name off the template (e.g. "Two_Of_Clubs" and "Ace_Of_Spades" ) at the path "Assets/Tilemaps/Blackjack/Playing_Cards/"
    public Sprite CardSprite => Resources.Load<Sprite>($"Blackjack/Playing_Cards/{Rank}_Of_{Suit}");
    public Card(Suit suit, Rank rank)
    {
        Suit = suit;
        Rank = rank;
}

    public int GetValue()
    {
        switch (Rank)
        {
            case Rank.Ace:
                return 11;
            case Rank.Two:
                return 2;
            case Rank.Three:
                return 3;
            case Rank.Four:
                return 4;
            case Rank.Five:
                return 5;
            case Rank.Six:
                return 6;
            case Rank.Seven:
                return 7;
            case Rank.Eight:
                return 8;
            case Rank.Nine:
                return 9;
            case Rank.Ten:
            case Rank.Jack:
            case Rank.Queen:
            case Rank.King:
            default:
                return 10;
        }
    }

    public override string ToString()
    {
        return $"The {GetValue()} of {Suit}";
    }
}
